header-verifier/src/lib.rs

```diff
@@ -201,19 +219,92 @@ pub trait Headerverifier: events::EventsModule + setup_phase::SetupPhaseModule {
         &self,
         _signature: &ManagedBuffer,
         _bridge_operations_hash: &ManagedBuffer,
-    ) -> bool {
-        true
+        bls_keys_bitmap: ManagedBuffer,
+        bls_pub_keys: &ManagedVec<ManagedBuffer>,
+    ) {
+        let _approving_validators =
+            self.get_approving_validators(&bls_keys_bitmap, bls_pub_keys.len());
+
+        // self.crypto().verify_bls_aggregated_signature(
+        //     approving_validators,
+        //     bridge_operations_hash,
+        //     signature,
+        // );
     }
 
-    fn is_signature_count_valid(&self, pub_keys_count: usize) -> bool {
-        let total_bls_pub_keys = self.bls_pub_keys().len();
-        let minimum_signatures = 2 * total_bls_pub_keys / 3;
+    fn get_bls_keys_by_id(
+        &self,
+        ids: MultiValueEncoded<BigUint<Self::Api>>,
+    ) -> ManagedVec<ManagedBuffer> {
+        let mut bls_keys = ManagedVec::new();
+
+        for id in ids.into_iter() {
+            bls_keys.push(
+                self.bls_keys_map(self.get_chain_config_address())
+                    .get(&id)
+                    .unwrap_or_else(|| sc_panic!(BLS_KEY_NOT_REGISTERED)),
+            );
+        }
 
-        pub_keys_count > minimum_signatures
+        bls_keys
+    }
+
+    fn get_approving_validators(
+        &self,
+        bls_keys_bitmap: &ManagedBuffer,
+        bls_keys_length: usize,
+    ) -> ManagedVec<ManagedBuffer> {
+        let mut padded_bitmap_byte_array = [0u8; 1024];
+        bls_keys_bitmap.load_to_byte_array(&mut padded_bitmap_byte_array);
+
+        let bitmap_byte_array = &padded_bitmap_byte_array[..bls_keys_length];
+
+        let mut approving_validators_bls_keys: ManagedVec<Self::Api, ManagedBuffer> =
+            ManagedVec::new();
+
+        for (index, has_signed) in bitmap_byte_array.iter().enumerate() {
+            if *has_signed == 1u8 {
+                approving_validators_bls_keys.push(
+                    self.bls_keys_map(self.get_chain_config_address())
```

Comment:
You need to get BLS key from `self.bls_pub_keys(epoch)`, have `epoch` as argument in this function